app/javascript/src/NewService/components/NewServiceForm.jsx

```diff
@@ -0,0 +1,145 @@
+// @flow
+
+import React, {useState, useEffect} from 'react'
+import {fetch as fetchPolyfill} from 'whatwg-fetch'
+import 'core-js/es6/promise'
+import {ServiceSourceForm, FormTemplate} from 'NewService'
+import type {SourceFormProps, FormTemplateProps} from 'NewService/types'
+
+import {createReactWrapper} from 'utilities/createReactWrapper'
+
+const BASE_PATH = '/p/admin/service_discovery'
+const NAMESPACES_URL = `${BASE_PATH}/projects.json`
+
+type Props = {
+  isServiceDiscoveryAccesible: boolean,
+  isServiceDiscoveryUsable: boolean,
+  serviceDiscoveryAuthenticateUrl: string,
+  providerAdminServiceDiscoveryServicesPath: string,
+  adminServicesPath: string
+}
+
+const fetchData = async (url: string, setError) => {
+  const dataItem = url === NAMESPACES_URL
+    ? 'projects'
+    : 'services'
+
+  return fetchPolyfill(url).then(response => {
```

Comment:
do we have types for the response? where is it defined its API?
I'm wondering because of the `response.ok` on the next line